core/vm/evm.go

```diff
@@ -463,7 +463,7 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 	// - the nonce is non-zero
 	// - the code is non-empty
 	// - the storage is non-empty
-	contractHash := evm.StateDB.GetCodeHash(address)
+	contractHash := evm.resolveCodeHash(address)
```

Comment:
For contract deployment, it could be initiated in two ways:

- transaction with nil to field;
- opCreate or opCreate2

All the approaches requires the *direct destination* has zero code and the deployment target is the *direct destination* itself; rather than the delegation.

It might happen that the destination has a delegation whose code is empty (delegation.code == null) and bypass this condition.

Although If the destination has a delegation, then nonce must be non-zero. But I feel the correct check should be GetCode here?

Reply:
You're correct, great catch. Will revert this change.